cli/src/semgrep/core_runner.py

```diff
@@ -518,14 +519,16 @@ class CoreRunner:
     def __init__(
         self,
         jobs: Optional[int],
-        deep: bool,
+        engine: EngineType,
         timeout: int,
         max_memory: int,
         timeout_threshold: int,
         optimizations: str,
         core_opts_str: Optional[str],
     ):
-        self._jobs = jobs if jobs else 1 if deep else get_cpu_count()
+        self._jobs = (
+            jobs if jobs else 1 if engine is EngineType.DEEP_INTER else get_cpu_count()
```

Comment:
not a huge fan of this postfix ifthenelse.

Reply:
I agree that in this case, with two of those nested, it is more clear to use a separate if-statement.